web/frontend/src/pages/ballot/Show.tsx

```diff
@@ -9,32 +10,41 @@ import { ROUTE_BALLOT_INDEX } from '../../Routes';
 import useElection from 'components/utils/useElection';
 import usePostCall from 'components/utils/usePostCall';
 import { ENDPOINT_EVOTING_CAST_BALLOT } from 'components/utils/Endpoints';
-import Modal from 'components/modal/Modal';
 import { OPEN } from 'components/utils/StatusNumber';
 import { encryptVote } from './components/VoteEncrypt';
-import './Show.css';
+import { ballotIsValid } from './components/HandleAnswers';
+import { handleOnDragEnd } from './components/RankDisplay';
+import { voteEncode } from './components/VoteEncode';
+import useConfiguration from 'components/utils/useConfiguration';
+import { ID, Question, RANK, ROOT_ID, SELECT, SUBJECT, TEXT } from 'types/configuration';
+import { DragDropContext } from 'react-beautiful-dnd';
+//import BallotCast from './BallotCast';
+import ModalNav from 'components/modal/RedirectToModal';
 
 const Ballot: FC = () => {
-  var bytes: number[];
   const { t } = useTranslation();
   const { electionId } = useParams();
   const token = sessionStorage.getItem('token');
-  const { loading, title, candidates, electionID, status, pubKey } = useElection(electionId, token);
-  const [choice, setChoice] = useState('');
-  const [userErrors, setUserErrors] = useState({ noCandidate: '' });
+  const { loading, configuration, electionID, status, pubKey, ballotSize } = useElection(
+    electionId,
+    token
+  );
+  const { sortedQuestions, answers, setAnswers } = useConfiguration(configuration);
+  const [userErrors, setUserErrors] = useState('');
   const edCurve = kyber.curve.newCurve('edwards25519');
   const [postRequest, setPostRequest] = useState(null);
   const [postError, setPostError] = useState('');
-  const { postData } = usePostCall(setPostError);
+  //const [showFlash, setShowFlash] = useState(false);
   const [showModal, setShowModal] = useState(false);
   const [modalText, setModalText] = useState(t('voteSuccess') as string);
+  const { postData } = usePostCall(setPostError);
 
   useEffect(() => {
     if (postRequest !== null) {
-      setPostError('');
+      //postData(ENDPOINT_EVOTING_CAST_BALLOT, postRequest, setShowFlash);
```

Comment:
yep, we might in the end use modal instead of flash. If that's the case then commented code should be removed.